GVFS/GVFS.Common/Git/DiffTreeResult.cs

```diff
@@ -75,7 +75,8 @@ public static DiffTreeResult ParseFromDiffTreeLine(string line, string repoRoot)
             // Take the mode, sha, operation part and split on a space then add the paths that were split on a tab to the end
             parts = parts[0].Split(' ').Concat(parts.Skip(1)).ToArray();
 
-            if (parts.Length != 6)
+            if (parts.Length != 6 ||
+                parts[5].Contains('\t'))

```

Comment:
Previously we were relying on a later call to `Path.Combine` to throw if there is a `\t` in the path, but that is brittle and not cross-platform. This check makes the condition explicit.

Reply:
Just adding there is already a unit test for this case (ParseFromDiffTreeLine_TwoPathLine) and the associated comment is updated to reflect the change.

Reply:
Are there other invalid path characters that we should be checking for here or do we still rely on `Path.Combine` to catch those?

Reply:
@jeschu1 yep this was the code change that enabled that unit test to be re-enabled.

@kewillford I don't know, this is all that was required to make the test pass. In reality, the test is not 100% sound, it's just trying to confirm that we're not calling `git diff-tree` with the wrong parameters, but it's doing it in a really round-about way.